esx_service/cli/vmdkops_admin.py

```diff
@@ -563,11 +563,14 @@ def get_creation_info(metadata):
 
 
 def get_attached_to(metadata):
-    """ Return which VM a volume is attached to based on its metadata """
+    """ Return which VM a volume is attached to based on its metadata. """
     try:
         vm_name = vmdk_ops.vm_uuid2name(metadata[kv.ATTACHED_VM_UUID])
         if not vm_name:
+            if metadata[kv.ATTACHED_VM_UUID]:
+                return metadata[kv.ATTACHED_VM_UUID]
```

Comment:
VM name was remvoed earlier to avoid the issue with a VM's name changing, which is why the name is retrieved each time. I'd vote to change the header to VM (name/uuid) for this issue.